runtime/near-vm-logic/src/logic.rs

```diff
@@ -49,7 +49,7 @@ pub struct VMLogic<'a> {
     invalid_iterators: HashSet<IteratorIndex>,
 
     /// The DAG of promises, indexed by promise id.
-    promises: HashSet<Promise>,
+    promises: Vec<Promise>,
```

Comment:
Why not store `Vec<PromiseToReceipt>` here directly. `promise_idx` always matches the index of the Promise in the vec

Reply:
`Promise` struct will expand later to support additional information about the constructed DAG so that we can do computation of fees. I've added it before but then had to undo since we will be adding fees after the merge.

Reply:
Got it. But can `promise_idx` point to non-index at that moment?

Reply:
Fixed.